trace/span.proto

```diff
@@ -0,0 +1,323 @@
+  string message = 2;
+
+  // A list of messages that carry the error details.  There is a common set of
+  // message types for APIs to use.
+  repeated google.protobuf.Any details = 3;
+}
+
+// The allowed types for [VALUE] in a `[KEY]:[VALUE]` attribute.
+message AttributeValue {
+  // The type of the value.
+  oneof value {
+    // A string up to 256 bytes long.
+    TruncatableString string_value = 1;
+    // A 64-bit signed integer.
+    int64 int_value = 2;
+    // A Boolean value represented by `true` or `false`.
+    bool bool_value = 3;
+  }
+}
+
+// A call stack appearing in a trace.
+message StackTrace {
+  // Represents a single stack frame in a stack trace.
+  message StackFrame {
+    // The fully-qualified name that uniquely identifies the function or
+    // method that is active in this frame.
+    TruncatableString function_name = 1;
+    // An un-mangled function name, if `function_name` is
+    // [mangled](http://www.avabodh.com/cxxin/namemangling.html). The name can
+    // be fully-qualified.
+    TruncatableString original_function_name = 2;
+    // The name of the source file where the function call appears.
+    TruncatableString file_name = 3;
+    // The line number in `file_name` where the function call appears.
+    int64 line_number = 4;
+    // The column number where the function call appears, if available.
+    // This is important in JavaScript because of its anonymous functions.
+    int64 column_number = 5;
+    // The binary module from where the code was loaded.
+    Module load_module = 6;
+    // The version of the deployed source code.
+    TruncatableString source_version = 7;
+  }
+
+  // A collection of stack frames, which can be truncated.
+  message StackFrames {
+    // Stack frames in this call stack.
+    repeated StackFrame frame = 1;
+    // The number of stack frames that were dropped because there
+    // were too many stack frames.
+    // If this value is 0, then no stack frames were dropped.
+    int32 dropped_frames_count = 2;
+  }
+
+  // Stack frames in this stack trace.
+  StackFrames stack_frames = 1;
+
+  // The hash ID is used to conserve network bandwidth for duplicate
+  // stack traces within a single trace.
+  //
+  // Often multiple spans will have identical stack traces.
+  // The first occurrence of a stack trace should contain both the
+  // `stackFrame` content and a value in `stackTraceHashId`.
+  //
+  // Subsequent spans within the same request can refer
+  // to that stack trace by only setting `stackTraceHashId`.
+  uint64 stack_trace_hash_id = 2;
+}
+
+// Binary module.
+message Module {
+  // For example: main binary, kernel modules, and dynamic libraries
+  // such as libc.so, sharedlib.so.
+  TruncatableString module = 1;
+
+  // A unique identifier for the module, usually a hash of its
+  // contents.
+  TruncatableString build_id = 2;
+}
+
+// Represents a string that might be shortened to a specified length.
```

Comment:
interesting as this hints that truncation will occur at origin.. or... a decorating proxy can also do this, too I suppose (just thinking)

Reply:
it can happen at any point (library, proxy, backend). The reason why we have it here is mainly because it can happen in the library.